Fix #6883 relative welcome redirect (#6886)

gregw opened the pull request:
Cheery pick 9 to 10

Fix #6883 relative welcome redirect
 + make all redirects able to be relative
 + added test for relative redirection in ResourceService

Signed-off-by: Greg Wilkins <gregw@webtide.com>

gregw committed:
Fix #6883 relative welcome redirect (#6886)

Fix #6883 relative welcome redirect
 + make all redirects able to be relative
 + added test for relative redirection in ResourceService

Signed-off-by: Greg Wilkins <gregw@webtide.com>

joakime commented:
Need to fix the license header

joakime commented on tests/jetty-http-tools/src/main/java/org/eclipse/jetty/http/tools/matchers/HttpFieldsHeaderValue.java:
```diff
@@ -0,0 +1,60 @@
+//  and Apache License v2.0 which accompanies this distribution.
+//
+//      The Eclipse Public License is available at
+//      http://www.eclipse.org/legal/epl-v10.html
```
This is the wrong license header for Jetty 10.0.x

joakime reviewed (changes requested):
License header needs to be fixed

gregw committed:
Fix #6860 IPv6 Format (#6861)

Fix #6860 IPv6 format by adding an extensible HttpChannel method

Signed-off-by: Greg Wilkins <gregw@webtide.com>
Co-authored-by: Lachlan Roberts <lachlan@webtide.com>
Signed-off-by: Greg Wilkins <gregw@webtide.com>
# Conflicts:
#	jetty-server/src/main/java/org/eclipse/jetty/server/HttpChannel.java
#	jetty-server/src/main/java/org/eclipse/jetty/server/Request.java
#	jetty-server/src/main/java/org/eclipse/jetty/server/handler/ContextHandler.java
#	jetty-server/src/test/java/org/eclipse/jetty/server/RequestTest.java

gregw committed:
Fix #6860 IPv6 Format (#6861)

Fix #6860 IPv6 format by adding an extensible HttpChannel method

Signed-off-by: Greg Wilkins <gregw@webtide.com>
Co-authored-by: Lachlan Roberts <lachlan@webtide.com>
Signed-off-by: Greg Wilkins <gregw@webtide.com>
# Conflicts:
#	jetty-server/src/main/java/org/eclipse/jetty/server/HttpChannel.java
#	jetty-server/src/main/java/org/eclipse/jetty/server/Request.java
#	jetty-server/src/main/java/org/eclipse/jetty/server/handler/ContextHandler.java
#	jetty-server/src/test/java/org/eclipse/jetty/server/RequestTest.java
